homeassistant/components/tplink/config_flow.py

```diff
@@ -45,7 +45,16 @@ async def async_step_integration_discovery(
 
     async def _async_handle_discovery(self, host: str, mac: str) -> FlowResult:
         """Handle any discovery."""
-        await self.async_set_unique_id(dr.format_mac(mac))
+        formatted_mac = dr.format_mac(mac)
+        if current_entry := await self.async_set_unique_id(formatted_mac):
+            if (
+                current_entry.state is config_entries.ConfigEntryState.SETUP_RETRY
+                and current_entry.data[CONF_HOST] == host
+            ):
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(current_entry.entry_id)
```

Comment:
Feels like we should be able to do this automatically for all integrations with discovery flows since we set a unique id of the config entry and flow.

Reply:
Yeah that makes more sense.  I'll prepare an alternate PR

Reply:
https://github.com/home-assistant/core/pull/72738